tuf/tuf.go

```diff
@@ -185,10 +211,19 @@ func (tr *Repo) UpdateDelegations(role *data.Role, keys []data.PublicKey) error
 		return data.ErrInvalidRole{Role: role.Name, Reason: "not a valid delegated role"}
 	}
 	parent := filepath.Dir(role.Name)
-	p, ok := tr.Targets[parent]
-	if !ok {
+
+	// check the parent role
+	if parentRole := tr.keysDB.GetRole(parent); parentRole == nil {
 		return data.ErrInvalidRole{Role: role.Name, Reason: "parent role not found"}
 	}
+
+	// check the parent role's metadata
+	p, ok := tr.Targets[parent]
+	if !ok { // the parent targetfile may not exist yet - if not, then create it
+		p = data.NewTargets()
```

Comment:
Hmm... I wouldn't mind doing so - we can check for signing keys before updating any delegations at all.  Would that be better than adding and letting the signing fail?